config/notifiers.go

```diff
@@ -295,6 +310,39 @@ func (c *WebhookConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return checkOverflow(c.XXX, "webhook config")
 }
 
+// WechatConfig configures notifications via Wechat.
+type WechatConfig struct {
+	NotifierConfig `yaml:",inline" json:",inline"`
+
+	ApiSecret string `yaml:"api_secret,omitempty" json:"api_secret,omitempty"`
+	CorpID    string `yaml:"corp_id,omitempty" json:"corp_id,omitempty"`
+	Message   string `yaml:"message,omitempty" json:"message,omitempty"`
+	ApiURL    string `yaml:"api_url,omitempty" json:"api_url,omitempty"`
+	ToUser    string `yaml:"touser,omitempty" json:"touser,omitempty"`
+	ToParty   string `yaml:"toparty,omitempty" json:"toparty,omitempty"`
+	Totag     string `yaml:"totag,omitempty" json:"totag,omitempty"`
+	AgentID   string `yaml:"agent_id,omitempty" json:"agent_id,omitempty"`
+
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline" json:"-"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *WechatConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultWechatConfig
+	type plain WechatConfig
+	if err := unmarshal((*plain)(c)); err != nil {
+		return err
+	}
+	if c.ApiSecret == "" {
+		return fmt.Errorf("missing Wechat ApiSecret in Wechat config")
```

Comment:
This can be tested. see https://github.com/prometheus/alertmanager/blob/master/config/notifiers_test.go#L147